tests/unit_tests/cuda/CMakeLists.txt

```diff
@@ -0,0 +1,19 @@
+add_library(cuda_test SHARED
+  grids_grid2_cuda_kernel.cu)
+
+target_include_directories(cuda_test
+  PUBLIC
+  ${CMAKE_CUDA_TOOLKIT_INCLUDE_DIRECTORIES}
+  )
+
+target_compile_options(cuda_test
+  PUBLIC $<$<COMPILE_LANGUAGE:CUDA>:
+  #-arch=sm_52 
+  #-gencode=arch=compute_75,code=sm_75 
+  --expt-relaxed-constexpr>)
+  #--use_fast_math>)
+  #-fmad=false>)
+
+target_link_libraries(cuda_test PUBLIC detray::core vecmem::cuda detray::tests_common)
```

Comment:
I think it would be better to not implement the `main(...)` function in the test yourself, but rather just link against `GTest::gtest_main`. Like we do it in VecMem.

https://github.com/acts-project/vecmem/blob/main/tests/cuda/CMakeLists.txt#L22

This way we don't have to update the code when/if GTest changes its "main function recommendations".

Reply:
Yup. I changed `add_detray_test` similar with vecmem test function